octobot_trading/personal_data/orders/order.py

```diff
@@ -604,6 +609,56 @@ async def update_from_order(self, other_order):
+        order_dict = order_details.get(orders_storage.OrdersStorage.ORIGIN_VALUE_KEY, {})
+        self.tag = order_dict.get(enums.ExchangeConstantsOrderColumns.TAG.value, self.tag)
+        self.trader_creation_kwargs = order_details.get(enums.StoredOrdersAttr.TRADER_CREATION_KWARGS.value,
+                                                        self.trader_creation_kwargs)
+        self.exchange_creation_params = order_details.get(enums.StoredOrdersAttr.EXCHANGE_CREATION_PARAMS.value,
+                                                          self.exchange_creation_params)
+        self.set_shared_signal_order_id(order_details.get(enums.StoredOrdersAttr.SHARED_SIGNAL_ORDER_ID.value,
+                                                          self.shared_signal_order_id))
+        self.has_been_bundled = order_details.get(enums.StoredOrdersAttr.HAS_BEEN_BUNDLED.value,
+                                                  self.has_been_bundled)
+        self.associated_entry_ids = order_details.get(enums.StoredOrdersAttr.ENTRIES.value,
+                                                      self.associated_entry_ids)
+
+    def _restore_order_group_from_storage_order_details(self, order_details, exchange_manager, pending_groups):
+        group = order_details.get(enums.StoredOrdersAttr.GROUP.value, None)
+        if group:
+            try:
+                group_name = group.get(enums.StoredOrdersAttr.GROUP_ID.value, None)
+                if group_name and group_name not in pending_groups:
+                    pending_groups[group_name] = \
+                        exchange_manager.exchange_personal_data.orders_manager.get_or_create_group(
+                            group_util.get_group_class(group[enums.StoredOrdersAttr.GROUP_TYPE.value]),
+                            group_name,
+                        )
+                self.order_group = pending_groups[group_name]
+            except KeyError as err:
+                logging.get_logger(self.get_logger_name()).error(f"Unhandled group type: {err}")
+
+    async def _restore_chained_orders_from_storage_order_details(self, order_details, exchange_manager, pending_groups):
+        chained_orders = order_details.get(enums.StoredOrdersAttr.CHAINED_ORDERS.value, None)
+        if chained_orders:
+            for chained_order in chained_orders:
+                chained_order_inst = await order_factory.create_order_from_order_storage_details(
+                    chained_order, exchange_manager, pending_groups
+                )
+                await chained_order_inst.set_as_chained_order(
+                    self,
+                    chained_order.get(enums.StoredOrdersAttr.HAS_BEEN_BUNDLED.value, False),
+                    chained_order.get(enums.StoredOrdersAttr.EXCHANGE_CREATION_PARAMS.value, {}),
+                    **chained_order.get(enums.StoredOrdersAttr.TRADER_CREATION_KWARGS.value, {}),
+                )
+                self.add_chained_order(chained_order_inst)
```

Comment:
I'm moving _restore_chained_orders_from_storage_order_details to order_factory, this will also remove the import in order.py.
I'll move _restore_order_group_from_storage_order_details to group_factory